x/dex/keeper/core.go

```diff
@@ -314,73 +323,53 @@ func (k Keeper) Swap0to1(goCtx context.Context, msg *types.MsgSwap, token0 strin
 		// Saving all the time
 		liq.Save(goCtx, k)
 		if initedTick != nil {
-			pair = InitLiquidityToken0(pair, initedTick.TickIndex)
+			pair.InitLiquidity(initedTick.TickIndex, is0To1)
```

Comment:
This approach of passing a boolean flag in describing which kind of liquidity to deal with is not very satisfying. IMO it would be better to use `LiquidityLocation` as discussed previously as parameters to this fn.